packages/core/testing/src/defer.ts

```
@@ -0,0 +1,52 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {ɵDeferBlockDetails as DeferBlockDetails, ɵDeferBlockInstanceState as DeferBlockInstanceState, ɵgetDeferBlocks as getDeferBlocks, ɵrenderDeferBlockState as renderDeferBlockState, ɵtriggerResourceLoading as triggerResourceLoading} from '@angular/core';
+import {CONTAINER_HEADER_OFFSET} from '@angular/core/src/render3/interfaces/container';
+
+import {type ComponentFixture} from './component_fixture';
+
+/**
+ * Represents an individual `{#defer}` block for testing purposes.
+ *
+ * @publicApi
+ * @developerPreview
+ */
+export class DeferBlockFixture {
+  constructor(
+      private block: DeferBlockDetails, private componentFixture: ComponentFixture<unknown>) {}
+
+  /**
+   * Renders the specified state of the defer fixture.
+   * @param state the defer state to render
+   */
+  async render(state: DeferBlockInstanceState): Promise<void> {
+    if (state === DeferBlockInstanceState.COMPLETE) {
+      await triggerResourceLoading(this.block.details, this.block.lView);
+    }
+    renderDeferBlockState(state, this.block.tNode, this.block.lContainer);
```

We should consider a scenario where we try to render a block in a state which doesn't have a corresponding template, e.g. we are trying to render a block in a loading state, but there was no `{:loading}` block defined. I think we should throw an error in this case.

---

I've added logic to throw in this case. PTAL.